### PR TITLE
Link to checkseum.c where needed

### PR DESCRIPTION
**Pull request by zshipko**
I'm not sure if this is totally right, but I needed to add this to get Irmin to build after upgrading some opam packages.

### CODE REVIEW

**Comment by dinosaure** [comment]
Hmmhmm, I think the situation it's because we just merge mirage/ocaml-git#303 which use the last version of `decompress`. Then, because dependency on `checkseum.{c,ocaml}` is not transitive, indeed, you need to add it in front of `git-{unix,mirage}`.

However, I'm not sure than you need to do this to the `irmin-git` package which is a library, and by this way, don't need to link to the `checkseum` implementation. In other words, `checkseum.{c,ocaml}` needs to be linked only on executable or libraries which you want to enforce one of these implementations - as you can see, [`git` does not need to be linked with `checkseum`](https://github.com/mirage/ocaml-git/blob/master/src/git/jbuild).

**Comment by zshipko** [comment]
I see, so this should just be added to `irmin-unix` instead? In that case, I think we should include a note about this requirement somewhere in the documentation for `irmin-git` too.

**Comment by samoht** [comment]
Note: this situation is transitory and should be improved when `dune` will support [build variants](https://github.com/ocaml/dune/issues/136).

File: test/irmin-git/jbuild [hunk 1]
```diff
@@ -3,7 +3,7 @@
 (library
  ((name      test_git)
   (modules   (test_git))
-  (libraries (irmin-test irmin-git git-unix))))
+  (libraries (irmin-test checkseum.c irmin-git git-unix))))
```
Comment (samoht) [inline test/irmin-git/jbuild]: this should not be needed, e.g. `checkseum` the concrete implementation (C or OCaml) should be chosen at link time only.

Comment (zshipko) [reply test/irmin-git/jbuild]: Without this the `irmin-git` tests wouldn’t build though. What is the preferred way of handling this?

**Comment by zshipko** [comment]
Okay, glad to hear there’s a more general fix planned but it looks like build variants are still being worked on heavily. What should we do in the meantime? I have no problem just keeping these changes in my local repo if this isn’t something we should worry about right now.

File: test/irmin-git/jbuild (hunk 1 above)
Comment (samoht) [reply test/irmin-git/jbuild]: You should add `checkseum.c` to all built executable, e.g. the test binaries too.

Comment (zshipko) [reply test/irmin-git/jbuild]: Got it, I've moved the linking to `checkseum.c` into the executable sections of the necessary jbuild files. Everything seems to be building correctly now (including tests and examples) using the most recent commit.

**Pull request merged**